debezium-core/src/main/java/io/debezium/pipeline/ChangeEventSourceCoordinator.java

```diff
@@ -56,20 +56,20 @@
      */
     public static final Duration SHUTDOWN_WAIT_TIMEOUT = Duration.ofSeconds(90);
 
-    private final Offsets<P, O> previousOffsets;
-    private final ErrorHandler errorHandler;
-    private final ChangeEventSourceFactory<P, O> changeEventSourceFactory;
-    private final ChangeEventSourceMetricsFactory changeEventSourceMetricsFactory;
-    private final ExecutorService executor;
-    private final EventDispatcher<?> eventDispatcher;
-    private final DatabaseSchema<?> schema;
+    protected final Offsets<P, O> previousOffsets;
+    protected final ErrorHandler errorHandler;
+    protected final ChangeEventSourceFactory<P, O> changeEventSourceFactory;
+    protected final ChangeEventSourceMetricsFactory changeEventSourceMetricsFactory;
+    protected final ExecutorService executor;
+    protected final EventDispatcher<?> eventDispatcher;
+    protected final DatabaseSchema<?> schema;
 
-    private volatile boolean running;
-    private volatile StreamingChangeEventSource<P, O> streamingSource;
-    private final ReentrantLock commitOffsetLock = new ReentrantLock();
+    protected volatile boolean running;
```

Comment:
In this case I'd prefer to keep it private and provide a method that returns whether it is still running or not. Idon't think the subclasses should access it directly as it is really implementation detail and as it is a part accessed by multple threads it is better to really limit who can write to it..

Reply:
I've updated the commit with this change. Thanks again for the quick feedback!